airflow/models.py

```diff
@@ -3024,6 +3024,25 @@ def get_active_runs(self, session=None):
         return active_dates
 
     @provide_session
+    def get_num_active_runs(self, external_trigger=None, session=None):
```

Comment:
> Are you sure your count() function is faster: http://docs.sqlalchemy.org/en/latest/orm/query.html and https://stackoverflow.com/questions/14754994/why-is-sqlalchemy-count-much-slower-than-the-raw-query ?

I'm not claiming that the `count()` query would be faster, furthermore the performance of the query does not even depend on sqlalchemy but the query engine with the optimizations that it can do. This does reduce the payload on transfer.

> As performance indeed isn't a real issue. How many active dag runs will you find? 200? 400? 1000? I'm not particularly convinced it warrants the extra complexity of adding this function or not using DagRun.find.

Like you mentioned, query performance is not an issue, but this does reduce the payload transferred. I think this function does not add any more complexity than the one is already there. Additionally, like I mentioned before, it seems this method could be reused in other bits of the codebase; using `DagRun.find` would just duplicate code on every place this can be used.